blackduck/HubRestApi.py

```diff
@@ -123,6 +128,9 @@ def __init__(self, *args, **kwargs):
         
     def read_config(self):
         try:
+            #always return to the examples directory to read the config
+            if not os.getcwd().endswith("examples"):
+                os.chdir(root_dir)
```

Comment:
could this potentially create a problem if you were calling this from another application that uses the blackduck library, rather than from the examples directory?

Reply:
Hi, yes, this is a potential problem that I've been hoping to make progress on before merging. The root of the problem is that .restconfig is located in examples on the first HubRestApi object init. When a refresh actually does occur, .restconfig no longer exists in context at the time HubRestApi reinitializes itself, so I reset back to examples to ensure that .restconfig is available. There has to be a better approach, it's just not clear to me. Any ideas?

Reply:
I believe the .restonfig.json is written to (or read from) the _current_ working directory. It is not hard-coded to anything. So, if I invoke a program in /Users/gsnyder/Projects/sage for instance, the .restonfig.json needs to be in the /Users/gsnyder/Projects/sage directory for it to be read. Or, if I was initializing the HubInstance using parameters, then the .restconfig.json would be written to that directory. 

Also, the config is saved in memory to self.config, so do we really need to read anything again?

Reply:
I removed the hardcoded part. Now, it works under the assumption that .restconfig.json is in the current working directory.